docs/source/keywords.rst

```diff
@@ -217,10 +221,26 @@ Backend-specific options
     * * ``Kokkos_ENABLE_ATOMICS_BYPASS``
       * Disable atomics when no host parallel nor device backend is enabled for Serial only builds (since Kokkos 4.3)
       * ``OFF``
-       
-
+    * * ``Kokkos_ENABLE_IMPL_CUDA_MALLOC_ASYNC``
+      * Enable ``cudaMallocAsync`` (requires CUDA Toolkit 11.2). This optimization may
+	improve performance in applications with multiple CUDA streams per device, but it
+	is known to be incompatible with MPI distributions built on older versions of UCX
+	and many Cray MPICH instances. See `kokkos/kokkos#7353 <https://github.com/kokkos/kokkos/pull/7353>`_
+      * ``ON``
+	
+    * * ``Kokkos_ENABLE_IMPL_HPX_ASYNC_DISPATCH``
+      * Enable asynchronous dispatch for the HPX backend
+      * ``ON``
+	
 ``Kokkos_ENABLE_CUDA_LAMBDA`` default value is ``OFF`` until 3.7 and ``ON`` since 4.0
 
+``Kokkos_ENABLE_IMPL_CUDA_MALLOC_ASYNC`` default value is ``OFF`` until 4.2, and ``ON`` until 4.5
+
+Any CMake keywords containing `IMPL` can fundamentally alter the underlying implementation
+of Kokkos on a given backend. It is encouraged not to set these unless the user knows what
+they are doing can accept the behavior changing without warning.
```

Comment:
> can fundamentally alter the underlying implementation of Kokkos on a given backend

We should make it a general warning right after the note suggesting that users can use ccmake to discover the list of options.  It is not specific to backend options.  Essentially, anything that says `IMPL` is considered private implementation detail and we reserve the right to change at anytime without prior notice.

@nliber would you please add a section on CMake variables in the [compatibility guidelines](https://kokkos.org/kokkos-core-wiki/ProgrammingGuide/Compatibility.html)